src/plugins/badge.coffee

```diff
@@ -0,0 +1,620 @@
+  }
+
+  #sa-badge-embedded-footer #sa-badge-embedded-more-button:hover {
+    color: white !important;
+    background-color: #d8721c !important;
+  }
+
+  @media only screen and (-webkit-min-device-pixel-ratio: 2),
+         only screen and (min--moz-device-pixel-ratio: 2),
+         only screen and (  -o-min-device-pixel-ratio: 2/1),
+         only screen and (     min-device-pixel-ratio: 2),
+         only screen and (     min-resolution: 192dpi),
+         only screen and (     min-resolution: 2dppx) {
+
+    #sa-badge-embedded-header #sa-badge-embedded-logo {
+      background-image: url('#{asset_url("badge/embedded/theme/#{configuration.theme}/logo_@@flavor@2x.png")}');
+    }
+
+    #sa-badge-embedded-header #sa-badge-embedded-rating .stars-container .sa-badge-full-star {
+      background-image: url('#{asset_url("badge/embedded/theme/#{configuration.theme}/star_full@2x.png")}');
+    }
+
+    #sa-badge-embedded-header #sa-badge-embedded-rating .stars-container .sa-badge-half-star {
+      background-image: url('#{asset_url("badge/embedded/theme/#{configuration.theme}/star_half@2x.png")}');
+    }
+
+    #sa-badge-embedded-header #sa-badge-embedded-rating .stars-container .sa-badge-empty-star {
+      background-image: url('#{asset_url("badge/embedded/theme/#{configuration.theme}/star_empty@2x.png")}');
+    }
+  }
+
+  #sa-badge-modal #sa-badge-modal-inner #sa-badge-modal-iframe {
+    display: block;
+
+    position: relative;
+    z-index: 2;
+  }
+
+  #sa-badge-modal #sa-badge-modal-inner {
+    background-color: #f1f1f1;
+  }
+  """
+
+  FLOATING_TEMPLATE = (assigns) -> """
+  <div id="sa-badge-floating-stars-container">
+   #{(STAR_TEMPLATE(star_type, assigns) for star_type in assigns.stars).join("\n")}
+  </div>
+  """
+
+  EMBEDDED_TEMPLATE = (assigns) -> """
+  <div id="sa-badge-embedded-header">
+    <div id="sa-badge-embedded-logo"></div>
+    <div id="sa-badge-embedded-rating">
+      <div class="rating-number"><span>#{assigns.rating}</span></div>
+      <div class="stars-container">
+       #{(STAR_TEMPLATE(star_type, assigns) for star_type in assigns.stars).join("\n")}
+      </div>
+    </div>
+  </div>
+  <div id="sa-badge-embedded-footer">
+    <div id="sa-badge-embedded-more-button">@@translations.badge.more</div>
+  </div>
+  """
+
+  STAR_TEMPLATE = (type, assigns) -> """
+  <div class="sa-badge-star sa-badge-#{type}-star"></div>
+  """
+
+  MODAL_INNER_TEMPLATE = (assigns) -> """
+  <div id="sa-badge-modal-inner" class="sa-badge-spinner">
+    <span id="sa-badge-modal-close-button"></span>
+      <iframe id="sa-badge-modal-iframe" src="#{assigns.endpoint}" width="100%" height="100%" frameborder="0"></iframe>
+    </div>
+  </div>
+  """
+
+  constructor: ->
+    @_setParentDoc()
+
+    return unless @_validate()
+
+    @addStyle()
+    @onDOMReady => @initialize()
+
+  _validate: ->
+    if context().configuration.display == 'embedded'
+      if not @$embeddedBadgePlugin()
```

Comment:
I followed the pattern in OrderStash plugin where we don't memoize either. But perhaps we should perform memoization in those cases.

The validation make sure that nothing happens if the target div is not found in the DOM, when badge plugin ahs been configured as "embedded".